src/BTCPayServer.Lightning.Eclair/EclairLightningClient.cs

```diff
@@ -0,0 +1,284 @@
+                            return new PayResponse(PayResult.CouldNotFindRoute);
+                        case "PENDING":
+                            await Task.Delay(200, cancellation);
+                            break;
+                    }
+                }
+            }
+            catch (EclairClient.EclairApiException e)
+            {
+            }
+
+            return new PayResponse(PayResult.CouldNotFindRoute);
+        }
+
+        public async Task<OpenChannelResponse> OpenChannel(OpenChannelRequest openChannelRequest,
+            CancellationToken cancellation = default(CancellationToken))
+        {
+            try
+            {
+                var result = await _eclairClient.Open(openChannelRequest.NodeInfo.NodeId,
+                    openChannelRequest.ChannelAmount.Satoshi
+                    , null,
+                    Convert.ToInt64(openChannelRequest.FeeRate.SatoshiPerByte), null, cancellation);
+
+                if (result.Contains("created channel"))
+                {
+                    var channelId = result.Replace("created channel", "").Trim();
+                    var channel = await _eclairClient.Channel(channelId, cancellation);
+                    switch (channel.State)
+                    {
+                        case "WAIT_FOR_OPEN_CHANNEL":
+                        case "WAIT_FOR_ACCEPT_CHANNEL":
+                        case "WAIT_FOR_FUNDING_CREATED":
+                        case "WAIT_FOR_FUNDING_SIGNED":
+                        case "WAIT_FOR_FUNDING_LOCKED":
+                        case "WAIT_FOR_FUNDING_CONFIRMED":
+                            return new OpenChannelResponse(OpenChannelResult.NeedMoreConf);
+                    }
+                }
+
+                if (result.Contains("couldn't publish funding tx"))
+                {
+                    return new OpenChannelResponse(OpenChannelResult.CannotAffordFunding);
+                }
+
+                return new OpenChannelResponse(OpenChannelResult.Ok);
+            }
+            catch (Exception e)
+            {
+                if (e.Message.Contains("not connected") || e.Message.Contains("no connection to peer"))
+                {
+                    return new OpenChannelResponse(OpenChannelResult.PeerNotConnected);
+                }
+
+                if (e.Message.Contains("insufficient funds"))
+                {
+                    return new OpenChannelResponse(OpenChannelResult.CannotAffordFunding);
+                }
+                if (e.Message.Contains("peer sent error: 'Multiple channels unsupported'"))
+                {
+                    return new OpenChannelResponse(OpenChannelResult.AlreadyExists);
+                }
+                
+
+                return new OpenChannelResponse(OpenChannelResult.AlreadyExists);
+            }
+        }
+
+        public async Task<BitcoinAddress> GetDepositAddress()
+        {
+            return await _rpcClient.GetNewAddressAsync();
```

Comment:
wtf, they can't put this in their API? just because of it we need to pass the bitcoin node auth, this is not good.

Reply:
Ok what about this. We don't really need it for BTCPay, so we don't really care.
What about: If _rpcClient is null then throw `NotSupportedException`. 

Then, let's not document the bitcoin-host stuff as this will disappear once they add getdepositaddress to their API.
Just make sure that not passing RPC parameters should have everything else work fine.

In BTCPay we won't need to put the bitcoin conf stuff to the connection string.

Reply:
Done